File: app/models/fundraising_readiness.py
"""
Phase 8 -- Fundraising Readiness V1. Response contract only -- see
app/ai/fundraising_readiness.py's own module docstring for the full
design record (why this exists as a separate, deterministic assessment
rather than exposing the legacy readiness_score, and the exact formula
behind readiness_score below).

Its own file for the same reason app/models/founder_action.py is its own
file: Fundraising Readiness is a distinct concern from canonical Startup
Intelligence (SPS) -- this model never reuses startup_intelligence_score
or any Methodology v2 scoring type.
"""

from pydantic import BaseModel, Field


class PillarReadinessOut(BaseModel):
    pillar: str
    label: str
    score: float | None = None
    confidence: str
    evidence_coverage: float
    weight: float
    # 0-10 scale; None when the pillar has no score at all (Unavailable)
    # -- never a fabricated 0. See
    # app.ai.fundraising_readiness.compute_pillar_readiness()'s own
    # docstring for the exact formula this comes from.
    readiness_contribution: float | None = None
    top_strength: str | None = None
    top_weakness: str | None = None


class ReadinessGapOut(BaseModel):
    category: str
    pillar: str | None = None
    issue: str
    why_it_matters: str
    recommended_next_step: str
    # The exact text used as the founder_actions dedup/provenance key if
    # this gap is added to the Action Plan (source='fundraising_gap') --
    # see app/api.py's add-to-plan endpoint.
    source_text: str


class ChecklistItemOut(BaseModel):
    category: str
    status: str
    note: str


class FundraisingReadinessResponse(BaseModel):
    startup_id: int
    canonical_name: str
    has_canonical_analysis: bool
    stage_label: str
    stage_recognized: bool
    # 0-100, or None when has_canonical_analysis is False -- never a
    # fabricated score for an unanalyzed startup.
    readiness_score: float | None = None
    readiness_band: str | None = None
    pillar_readiness: list[PillarReadinessOut] = Field(default_factory=list)
    gaps: list[ReadinessGapOut] = Field(default_factory=list)
    investor_questions: list[str] = Field(default_factory=list)
    checklist: list[ChecklistItemOut] = Field(default_factory=list)
    has_pitch_deck: bool
    pitch_deck_note: str
    # Context only -- current SPS, shown so the page can explain the
    # distinction rather than pretend readiness doesn't relate to it at
    # all. Never blended into readiness_score.
    current_sps: float | None = None
    analyzed_at: str | None = None

File: dashboard/types/fundraisingReadiness.ts
// Phase 8 -- Fundraising Readiness V1. Mirrors
// app/models/fundraising_readiness.py exactly. This is a separate,
// deterministic assessment -- never SPS, never written to SPS history,
// never shown in Rankings/Compare. See
// app/ai/fundraising_readiness.py's own module docstring for the full
// design record.

export interface PillarReadiness {
  pillar: string;
  label: string;
  score: number | null;
  confidence: string;
  evidence_coverage: number;
  weight: number;
  readiness_contribution: number | null;
  top_strength: string | null;
  top_weakness: string | null;
}

export interface ReadinessGap {
  category: string;
  pillar: string | null;
  issue: string;
  why_it_matters: string;
  recommended_next_step: string;
  source_text: string;
}

export interface ChecklistItem {
  category: string;
  status: string;
  note: string;
}

export interface FundraisingReadiness {
  startup_id: number;
  canonical_name: string;
  has_canonical_analysis: boolean;
  stage_label: string;
  stage_recognized: boolean;
  readiness_score: number | null;
  readiness_band: string | null;
  pillar_readiness: PillarReadiness[];
  gaps: ReadinessGap[];
  investor_questions: string[];
  checklist: ChecklistItem[];
  has_pitch_deck: boolean;
  pitch_deck_note: string;
  current_sps: number | null;
  analyzed_at: string | null;
  // Phase 37D -- Unified Workspace Simplification + Legacy Containment.
  // Ownership-checked; non-null means this startup is linked to a
  // venture the caller owns, so the page redirects into that Venture
  // Workspace's Analyze tab instead of rendering here.
  linked_venture_id: number | null;
}
